cmd/ferretdb/main.go

```diff
@@ -111,20 +119,47 @@ func run() {
 		return
 	}
 
-	startFields := []zap.Field{
+	stateFile, err := filepath.Abs(filepath.Join(cli.StateDir, "state.json"))
+	if err != nil {
+		log.Fatalf("Failed to get path for state file: %s.", err)
+	}
+
+	p, err := state.NewProvider(stateFile)
```

Comment:
I like that part about providers and such things! It looks like something that could be extended in the future!